### PR TITLE
[autoscaler][docs] No more legacy autoscaler yamls

### PR DESCRIPTION
**Pull request by wuisawesome**
## Why are these changes needed?

This PR updates all autoscaler yaml examples/defaults to not use the legacy `head_node` and `worker_node` fields and deletes the explicit `example-full-legacy` yamls and the corresponding tests.

## Related issue number

For ease of review, this PR is purely cosmetic/yaml editing (plus minor test changes to keep CI happy).  It partially satisfies https://github.com/ray-project/ray/issues/20837. There will be 2 more follow up PRs (one to make the schema change and update the configs baked into unit tests, and another to clean up the legacy code). 

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by wuisawesome** [comment]
cc @scv119

**Comment by DmitriGekhtman** [comment]
Re: The planned follow-ups, I don't believe they are critical to the release (they are nice-to-have).
Main thing is to raise an error when the deprecated fields are present.

File: python/ray/autoscaler/kubernetes/example-ingress.yaml
Comment (DmitriGekhtman) [inline python/ray/autoscaler/kubernetes/example-ingress.yaml:112]: Would you mind deleting all of the configs in the kubernetes folder except for defaults.yaml?
(Optional, I can do it myself in a separate PR if needed.)

**Comment by DmitriGekhtman** [comment]
> Re: The planned follow-ups, I don't believe they are critical to the release (they are nice-to-have). Main thing is to raise an error when the deprecated fields are present.

I guess modifying the schema would raise the error on legacy fields.
...Just be warned that modifying all unit tests to use the new format might be painful.

Also...take a look at the examples in the "local" (on-prem) directory and make sure we don't screw those up -- those are a bit of a special case...

**Review by DmitriGekhtman — APPROVED** [review]

**Comment by AmeerHajAli** [comment]
Hey @wuisawesome , how confident are we that this change won't break things? Have we manually tested starting clusters on GCP and AWS and running an autoscaling application and verifying that it works?

**Comment by wuisawesome** [comment]
I will manually test the main configs before merging, and we should call on other maintainers to test the wheels too after the branch cut. We should be as confident as one can be without automated integration tests.

**Comment by wuisawesome** [comment]
In general, the process of updating these configs was fairly mechanical, so i think the risk shouldn't be super high.

**Comment by wuisawesome** [comment]
Manual confirmation aws example full still works. Will do more comprehensive testing after the other prs. 
image

**Comment by wuisawesome** [comment]
Mac agents have been waiting for agent for 2 hours. Everything else is green. Merging.

**Pull request merged**
